src/test/java/com/firebolt/jdbc/connection/settings/FireboltPropertiesTest.java

```diff
@@ -88,43 +88,54 @@ void shouldSupportIntForBooleanProperties() {
 	@Test
 	void shouldUseCustomPortWhenProvided() {
 		Properties properties = new Properties();
-		properties.put("path", "/example");
+		properties.put("path", "example");
 		properties.put("host", "host");
 		properties.put("port", "999");
 
 		assertEquals(999, FireboltProperties.of(properties).getPort());
 	}
 
 	@Test
-	void shouldThrowExceptionWhenNoDbProvided() {
+	void shouldUseSystemEngineWhenNoDbOrEngineProvided() {
 		Properties properties = new Properties();
-		properties.put("host", "host");
-
-		assertThrows(IllegalArgumentException.class, () -> FireboltProperties.of(properties));
+		FireboltProperties fireboltProperties = FireboltProperties.of(properties);
+		assertTrue(FireboltProperties.of(properties).isSystemEngine());
+		assertEquals("system", fireboltProperties.getEngine());
+		assertNull(fireboltProperties.getDatabase());
+		assertFalse(fireboltProperties.isCompress());
 	}
 
 	@Test
-	void shouldThrowExceptionWhenHostIsNotProvided() {
+	void shouldNotUseSystemEngineWhenDbAsPathIsProvided() {
 		Properties properties = new Properties();
-		assertThrows(IllegalArgumentException.class, () -> FireboltProperties.of(properties));
+		properties.put("path", "example");
+		FireboltProperties fireboltProperties = FireboltProperties.of(properties);
+		assertFalse(FireboltProperties.of(properties).isSystemEngine());
+		assertNull(fireboltProperties.getEngine());
+		assertEquals("example", fireboltProperties.getDatabase());
+		assertTrue(fireboltProperties.isCompress());
 	}
 
 	@Test
-	void shouldThrowExceptionWhenDbPathFormatIsInvalid() {
+	void shouldNotUseSystemEngineWhenDbAsQueryParamIsProvided() {
```

Comment:
I don't think this is correct. When db is not provided (but a user engine is) we fetch it via the system engine. If no engine and no db is provided we connect to a system engine with no db context.

Reply:
Let's review test later as a part of general refactoring and the code face-lifting.

Reply:
Clarified over the call, this is testing the Properties class and not the logic itself.